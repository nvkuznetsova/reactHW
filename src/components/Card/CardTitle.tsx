import React from 'react';

import { Badge } from 'src/components/Badge/Badge'

type CardTitleProps = {
  name: string,
  badgeText: string,
  badgeClassName: string,
};

export const CardTitle: React.FC<CardTitleProps> = ({ name, badgeText, badgeClassName }) => (
  <h4 className="card-title">
    {name}
    <Badge 
      badgeText={badgeText}
      badgeClassName={badgeClassName}
    />
  </h4>
);