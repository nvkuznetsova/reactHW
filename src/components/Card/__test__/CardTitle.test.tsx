import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';

import { CardTitle, CardTitleProps } from '../CardTitle';

describe('Card', () => {
  const getProps = (): CardTitleProps => ({
    name: 'test name',
    badgeText: 'test badge',
    badgeClassName: 'badge-class',
  });
  const setup = (): ShallowWrapper => shallow(<CardTitle {...getProps()} />);

  it('should match snapshot', () => {
    const component = setup();
    expect(component).toMatchSnapshot()
  });
});