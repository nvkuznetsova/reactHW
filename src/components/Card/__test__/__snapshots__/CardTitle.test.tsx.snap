// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match snapshot 1`] = `
<h4
  className="card-title"
>
  test name
  <Badge
    badgeClassName="badge-class"
    badgeText="test badge"
  />
</h4>
`;
